r1cs/gadgets/std/src/groups/curves/twisted_edwards/mod.rs

```diff
@@ -1302,11 +914,12 @@ mod projective_impl {
         ) -> Result<Self, SynthesisError>
         where
             FN: FnOnce() -> Result<T, SynthesisError>,
-            T: Borrow<TEProjective<P>>,
+            T: Borrow<TEExtended<P>>,
```

Comment:
A general comment: why do we enforce curve membership for an input?

Reply:
Probably a leftover, but we do it also for SW curves. Can we safely remove such checks ?